src/librbd/image/RemoveRequest.cc

```diff
@@ -621,6 +627,24 @@ void RemoveRequest<I>::handle_remove_v1_image(int r) {
   if (!m_old_format) {
     remove_v2_image();
   }
+
+}
+
+template<typename I>
+void RemoveRequest<I>::remove_image_cache() {
```

Comment:
I see, let me expain current design in this PR: if imageA on node A and imageB on node B share same parent snap, they won't share parent snap's cache since they're on different nodes, and remote cache access may have worse performance compared with achieve data from osd side.
Only imageA on nodes A and imageC on node A share same parent snap's cache. 
And when doing rbd remove action, "rbd rm imageA" will only remove imageA_cache, and it won't remove imageA's parent snap cache.
Do you think that makes sense to you.